src/foam/lib/json/PermissionedOutputter.java

```diff
@@ -0,0 +1,26 @@
+package foam.lib.json;
+
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+import foam.nanos.auth.AuthService;
+
+public class PermissionedOutputter extends Outputter {
+  protected foam.core.X x_;
+
+  public PermissionedOutputter(foam.core.X x, OutputterMode mode) {
+    super(mode);
+    this.x_ = x;
+  }
+
+  @Override
+  protected boolean propertyPredicate(FObject fo, PropertyInfo prop) {
+    if ( prop.getPermissionRequired() ) {
+      AuthService auth = (AuthService) this.x_.get("auth");
+      String propName = prop.toString();
```

Comment:
These next two lines generate a lot of garbage which will take more CPU to allocate and cleanup than the work you're doing. Couldn't you just do prop.getName()?